packages/zendesk-adapter/src/change_validators/custom_statuses_enabled.ts

```diff
@@ -60,6 +61,55 @@ const areCustomStatusesEnabled = async (
   return customStatusesEnabled.enabled
 }
 
+const createErrorsForCustomStatusTypes = (
+  changes: readonly Change<ChangeDataType>[]
+): ChangeError[] =>
+  changes
+    .filter(change => getChangeData(change).elemID.typeName === CUSTOM_STATUS_TYPE_NAME)
+    .map(getChangeData)
+    .map(instance => ({
+      elemID: instance.elemID,
+      severity: 'Error',
+      message: 'Custom statuses are not enabled.',
+      detailedMessage:
+        'Cannot deploy custom statuses when they are not enabled in the Zendesk account.',
+    }))
+
+const hasConditionWithCustomStatuses = (conditions: Condition[]): boolean =>
+  conditions.some((condition: Condition) =>
+    condition.child_fields.some(
+      field => (field.required_on_statuses.custom_statuses?.length ?? 0) > 0
```

Comment:
you can use `makeArray` from `lowerdash` -
```suggestion
      field => makeArray(field.required_on_statuses.custom_statuses).length > 0
```

Reply:
Done.